repos/system_upgrade/el8toel9/actors/checkcustomnetworkscripts/libraries/customnetworkscripts.py

```diff
@@ -0,0 +1,53 @@
+import os
+
+from leapp import reporting
+
+CUSTOM_NETWORK_SCRIPTS = [
+    "/sbin/ifup-local",
+    "/sbin/ifup-pre-local",
+    "/sbin/ifdown-local",
+    "/sbin/ifdown-pre-local",
+]
+DOC_URL = "https://red.ht/upgrading-RHEL-8-to-RHEL-9-network-scripts"
+
+
+def generate_report():
+    """ Generate reports informing user about possible manual intervention requirements """
+
+    # Show documentation url if custom network-scripts detected
+    title = "custom network-scripts detected"
+    summary = ("RHEL 9 does not support the legacy network-scripts package that was"
+               " deprecated in RHEL 8, and the custom network-scripts have been"
+               " detected. In order to proceed with the upgrade, you can move the"
+               " custom network-scripts, for example, to the `/opt/network-scripts/`"
+               " directory.")
+
+    reporting.create_report(
+        [
+            reporting.Title(title),
+            reporting.Summary(summary),
+            reporting.Remediation(
+                hint=(
+                    "Migrate the custom network-scripts to NetworkManager dispatcher"
+                    " scripts manually before the ugprade. Follow instructions in the"
+                    " official documentation."
+                )
+            ),
+            reporting.Severity(reporting.Severity.HIGH),
+            reporting.Flags([reporting.Flags.INHIBITOR]),
+            reporting.Tags([reporting.Tags.SERVICES]),
+            reporting.ExternalLink(
+                title=(
+                    "Upgrading from RHEL 8 to 9 - migrating custom network-scripts to"
+                    " NetworkManager dispatcher scripts"
+                ),
+                url=DOC_URL,
+            ),
+        ]
+        + [reporting.RelatedResource("file", fname) for fname in CUSTOM_NETWORK_SCRIPTS]
+    )
+
+
+def process():
+    if any(os.path.isfile(fname) for fname in CUSTOM_NETWORK_SCRIPTS):
```

Comment:
Hmm this way you will report all files specified in CUSTOM_NETWORK_SCRIPTS regardless of how many actually exist. Is it intended?
Maybe some additional os.path.isfile check should be added to generate_report, to filter out only actually existing files?

Reply:
Ideally, we should generate report as long as the system is detecting any one of the following scripts:
```
    "/sbin/ifup-local"
    "/sbin/ifup-pre-local"
    "/sbin/ifdown-local"
    "/sbin/ifdown-pre-local"
```

> Maybe some additional os.path.isfile check should be added to generate_report, to filter out only actually existing files?
what do you mean ?

Reply:
The generated report should ideally contain information relevant just to the actual system. What @fernflower means is that line 47 should look like this to print only relevant (present) resources:
```
        + [reporting.RelatedResource("file", fname) for fname in CUSTOM_NETWORK_SCRIPTS if os.path.exists(fname)]
```

So scripts that are not used on the system, are not reported.

Reply:
Ok, thanks for pointing out, I updated.

Reply:
There is no need to check the files twice if `generate_report()` can be modified to accept an argument (which it probably can):

```
def process():
    custom_network_scripts = [
    "/sbin/ifup-local",
    "/sbin/ifup-pre-local",
    "/sbin/ifdown-local",
    "/sbin/ifdown-pre-local",
]
    related_resources = [reporting.RelatedResource("file", filename) for filename in custom_network_scripts]
    if related_resources:
       generate_report(related_resources)
```

Reply:
@tyll it has to be Python3.6+ compatible as it's the default python for RHEL 8.